### PR TITLE
Adding JKAnime

### PR DESCRIPTION
**Pull request by Zerlingg**
Pause:
DiscordPTB_gwPGOR0ojK

Play:
DiscordPTB_H768CDog4m

### CODE REVIEW

**Comment by Zerlingg** [comment]
Oh an error, what do i have to do?

File: websites/J/JKAnime/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    "name": "Giingu",
+    "id": "156481332652802048"
+  },
+  "service": "JKAnime",
+  "description": {
+    "en": "JKAnime is a spanish anime streaming website with spanish subtitles.",
+    "zh_CN": "JKAnime是一个视频网站，您可以在其中观看带有西班牙语字幕的动漫。"
+  },
+  "url": "jkanime.net",
+  "version": "1.0",
```
Comment (Bas950) [inline websites/J/JKAnime/dist/metadata.json:12]: version 1.0.0

**Review by Bas950 — CHANGES_REQUESTED** [review]
Dont upload js files and fix deepscan issues.

**Review by doomlerd — CHANGES_REQUESTED** [review]
It would be better if you use the icons provided in the resources folder.

**Comment by Zerlingg** [comment]
Changed icons to the ones provided in resources folder and JS files deleted

**Comment by Zerlingg** [comment]
IDK how to fix the "SV / Schema Validation Check (pull_request)" thing

**Comment by kawakami** [comment]
You have incorrect version. It must be 1.0.0

**Comment by doomlerd** [comment]
> Changed icons to the ones provided in resources folder

Take and upload a screenshot of the change.

**Comment by Zerlingg** [comment]
image

**Comment by doomlerd** [comment]
I mean the presence itself

File: websites/J/JKAnime/presence.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+): Array<number> {
+  const startTime = Math.floor(Date.now());
+  const endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+  return [Math.floor(startTime / 1000), (Math.floor(endTime))];
+}
+
+presence.on(
+  "iFrameData",
+  (data: { duration: number; currentTime: number; paused: boolean }) => {
+    video = data;
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:27]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+};
+
+function getTimestamps(
+  videoTime: number,
+  videoDuration: number
+): Array<number> {
+  const startTime = Math.floor(Date.now());
+  const endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+  return [Math.floor(startTime / 1000), (Math.floor(endTime))];
+}
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:22]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  paused: true
+};
+
+function getTimestamps(
+  videoTime: number,
+  videoDuration: number
+): Array<number> {
+  const startTime = Math.floor(Date.now());
+  const endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+  return [Math.floor(startTime / 1000), (Math.floor(endTime))];
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:21]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+presence.on(
+  "iFrameData",
+  (data: { duration: number; currentTime: number; paused: boolean }) => {
+    video = data;
+  }
+);
+
+presence.on("UpdateData", async () => {
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:31]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+      video.duration
+    );
+    console.log("The current time is " + timestamps[0] + "and the durration is " + timestamps[1]);
+    console.log("The video is " + video.paused);
+    const title = document.querySelector(
+        "#reproductor-box > div.video-header > h1"
+      ).textContent;
+    if(title && timestamps[0] && timestamps[1]){
+        data.details = title;
+        (data.smallImageKey = video.paused ? "pause" : "play");
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:51]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  const data: PresenceData = {
+    largeImageKey: "jkanime"
+  };
+  console.log(video);
+  if (
+    video != null &&
+    !isNaN(video.duration)
+  ) {
+    const timestamps = getTimestamps(
+      video.currentTime,
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:41]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  });
+let video = {
+  duration: 0,
+  currentTime: 0,
+  paused: true
+};
+
+function getTimestamps(
+  videoTime: number,
+  videoDuration: number
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:17]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    browsing: "presence.activity.browsing"
+  });
+let video = {
+  duration: 0,
+  currentTime: 0,
+  paused: true
+};
+
+function getTimestamps(
+  videoTime: number,
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:16]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/iframe.ts [hunk 10]
```diff
@@ -0,0 +1,16 @@
+const iframe = new iFrame();
+
```
Comment (SeMiD) [inline websites/J/JKAnime/iframe.ts:2]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/iframe.ts [hunk 11]
```diff
@@ -0,0 +1,16 @@
+const iframe = new iFrame();
+
+iframe.on("UpdateData", async () => {
+
+	if(document.querySelector("video[id$='jkvideo_html5_api']") != null || document.querySelector("div.jw-media.jw-reset > video") != null) {
+		const video: HTMLVideoElement = document.querySelector("video[id$='jkvideo_html5_api']") != null
+		? document.querySelector("video[id$='jkvideo_html5_api']") : document.querySelector("div.jw-media.jw-reset > video");
+		if(video != null && !isNaN(video.duration)) {
+			iframe.send({
```
Comment (SeMiD) [inline websites/J/JKAnime/iframe.ts:9]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+    !isNaN(video.duration)
+  ) {
+    const timestamps = getTimestamps(
+      video.currentTime,
+      video.duration
+    );
+    console.log("The current time is " + timestamps[0] + "and the durration is " + timestamps[1]);
+    console.log("The video is " + video.paused);
+    const title = document.querySelector(
+        "#reproductor-box > div.video-header > h1"
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:47]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+presence.on("UpdateData", async () => {
+  const data: PresenceData = {
+    largeImageKey: "jkanime"
+  };
+  console.log(video);
+  if (
+    video != null &&
+    !isNaN(video.duration)
+  ) {
+    const timestamps = getTimestamps(
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:40]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+      video.currentTime,
+      video.duration
+    );
+    console.log("The current time is " + timestamps[0] + "and the durration is " + timestamps[1]);
+    console.log("The video is " + video.paused);
+    const title = document.querySelector(
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:46]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+);
+
+presence.on("UpdateData", async () => {
+  const data: PresenceData = {
+    largeImageKey: "jkanime"
+  };
+  console.log(video);
+  if (
+    video != null &&
+    !isNaN(video.duration)
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:38]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+  videoTime: number,
+  videoDuration: number
+): Array<number> {
+  const startTime = Math.floor(Date.now());
+  const endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+  return [Math.floor(startTime / 1000), (Math.floor(endTime))];
+}
+
+presence.on(
+  "iFrameData",
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:25]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 17]
```diff
@@ -0,0 +1,75 @@
+const presence = new Presence({
+    clientId: "731164832426098748"
+  }),
+  strings = presence.getStrings({
+    play: "presence.playback.playing",
+    pause: "presence.playback.paused",
+    browsing: "presence.activity.browsing"
+  });
+let video = {
+  duration: 0,
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:10]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 18]
```diff
@@ -0,0 +1,75 @@
+const presence = new Presence({
+    clientId: "731164832426098748"
+  }),
+  strings = presence.getStrings({
+    play: "presence.playback.playing",
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:5]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/iframe.ts [hunk 19]
```diff
@@ -0,0 +1,16 @@
+const iframe = new iFrame();
+
+iframe.on("UpdateData", async () => {
```
Comment (SeMiD) [inline websites/J/JKAnime/iframe.ts:3]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+  currentTime: 0,
+  paused: true
+};
+
+function getTimestamps(
+  videoTime: number,
+  videoDuration: number
+): Array<number> {
+  const startTime = Math.floor(Date.now());
+  const endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:20]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+  return [Math.floor(startTime / 1000), (Math.floor(endTime))];
+}
+
+presence.on(
+  "iFrameData",
+  (data: { duration: number; currentTime: number; paused: boolean }) => {
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:26]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/iframe.ts [hunk 22]
```diff
@@ -0,0 +1,16 @@
...
+
+	if(document.querySelector("video[id$='jkvideo_html5_api']") != null || document.querySelector("div.jw-media.jw-reset > video") != null) {
+		const video: HTMLVideoElement = document.querySelector("video[id$='jkvideo_html5_api']") != null
+		? document.querySelector("video[id$='jkvideo_html5_api']") : document.querySelector("div.jw-media.jw-reset > video");
+		if(video != null && !isNaN(video.duration)) {
+			iframe.send({
+				duration: video.duration,
+				currentTime: video.currentTime,
+				paused: video.paused
+			});
```
Comment (SeMiD) [inline websites/J/JKAnime/iframe.ts:13]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/iframe.ts [hunk 23]
```diff
@@ -0,0 +1,16 @@
+const iframe = new iFrame();
+
+iframe.on("UpdateData", async () => {
+
+	if(document.querySelector("video[id$='jkvideo_html5_api']") != null || document.querySelector("div.jw-media.jw-reset > video") != null) {
+		const video: HTMLVideoElement = document.querySelector("video[id$='jkvideo_html5_api']") != null
+		? document.querySelector("video[id$='jkvideo_html5_api']") : document.querySelector("div.jw-media.jw-reset > video");
+		if(video != null && !isNaN(video.duration)) {
```
Comment (SeMiD) [inline websites/J/JKAnime/iframe.ts:8]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 24]
```diff
@@ -0,0 +1,75 @@
...
+    video = data;
+  }
+);
+
+presence.on("UpdateData", async () => {
+  const data: PresenceData = {
+    largeImageKey: "jkanime"
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:33]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 25]
```diff
@@ -0,0 +1,75 @@
...
+  };
+  console.log(video);
+  if (
+    video != null &&
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:37]: Codacy found an issue: [Expected linebreaks to be 'LF' but found 'CRLF'.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

File: websites/J/JKAnime/presence.ts [hunk 26]
```diff
@@ -0,0 +1,75 @@
...
+  (data: { duration: number; currentTime: number; paused: boolean }) => {
+    video = data;
+  }
+);
+
+presence.on("UpdateData", async () => {
+  const data: PresenceData = {
+    largeImageKey: "jkanime"
+  };
+  console.log(video);
```
Comment (SeMiD) [inline websites/J/JKAnime/presence.ts:35]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/PreMiD/Presences/pullRequest?prid=6113512)

**Review by Bas950 — CHANGES_REQUESTED** [review]
Ignore deepscan and fix codacy issues.

**Comment by doomlerd** [comment]
Closing due to inactivity.

**Pull request closed without merging**